--- Smarty-2.6.12/libs/Smarty.class.php	2006-01-18 10:52:24.000000000 -0800
+++ smarty/Smarty.class.php	2006-01-23 15:07:45.000000000 -0800
@@ -30,7 +30,7 @@
  * @version 2.6.12
  */
 
-/* $Id$ */
+/* $Id$ */
 
 /**
  * DIR_SEP isn't used anymore, but third party apps might
@@ -1107,6 +1107,18 @@
     {
         $this->fetch($resource_name, $cache_id, $compile_id, true);
     }
+    
+    /**
+     * compiles a template without executing it
+     *
+     * @param string $resource_name
+     * @param string $cache_id
+     * @param string $compile_id
+     */
+    function compile($resource_name, $cache_id = null, $compile_id = null)
+    {
+        $this->fetch($resource_name, $cache_id, $compile_id, false, false);
+    }
 
     /**
      * executes & returns or displays the template results
@@ -1116,7 +1128,7 @@
      * @param string $compile_id
      * @param boolean $display
      */
-    function fetch($resource_name, $cache_id = null, $compile_id = null, $display = false)
+    function fetch($resource_name, $cache_id = null, $compile_id = null, $display = false, $execute = true)
     {
         static $_cache_info = array();
         
@@ -1262,7 +1274,7 @@
             if ($this->_is_compiled($resource_name, $_smarty_compile_path)
                     || $this->_compile_resource($resource_name, $_smarty_compile_path))
             {
-                include($_smarty_compile_path);
+            	if ($execute) { include($_smarty_compile_path); }
             }
             $_smarty_results = ob_get_contents();
             ob_end_clean();
