--- adodb462/drivers/adodb-postgres64.inc.php	Thu May  5 22:28:57 2005
+++ adodb/drivers/adodb-postgres64.inc.php	Thu May  5 21:41:28 2005
@@ -91,8 +91,8 @@
 	var $hasAffectedRows = true;
 	var $hasLimit = false;	// set to true for pgsql 7 only. support pgsql/mysql SELECT * FROM TABLE LIMIT 10
 	// below suggested by Freek Dijkstra 
-	var $true = 't';		// string that represents TRUE for a database
-	var $false = 'f';		// string that represents FALSE for a database
+	var $true = '1';		// string that represents TRUE for a database
+	var $false = '0';		// string that represents FALSE for a database
 	var $fmtDate = "'Y-m-d'";	// used by DBDate() as the default date format used by the database
 	var $fmtTimeStamp = "'Y-m-d G:i:s'"; // used by DBTimeStamp as the default timestamp fmt.
 	var $hasMoveFirst = true;
@@ -159,6 +159,22 @@
 		// to really return the id, we need the table and column-name, else we can only return the oid != id
 		return empty($table) || empty($column) ? $oid : $this->GetOne("SELECT $column FROM $table WHERE oid=".(int)$oid);
 	}
+	
+	// Added 2004-06-27 by Kevin Jamieson (http://www.pkp.ubc.ca/)
+	// Insert_ID function that returns the actual field value instead of the OID
+	function PO_Insert_ID($table="", $id="") {
+		if (!empty($table) && !empty($id)) {
+			$result = @pg_exec("SELECT CURRVAL('{$table}_{$id}_seq')");
+			if ($result) {
+				$row = @pg_fetch_row($result, 0);
+				if (isset($row[0])) {
+					return $row[0];
+				}
+			}
+		}
+		
+		return $this->_insertid();
+	}	
 
 // I get this error with PHP before 4.0.6 - jlim
 // Warning: This compilation does not support pg_cmdtuples() in d:/inetpub/wwwroot/php/adodb/adodb-postgres.inc.php on line 44
@@ -625,7 +641,8 @@
 		   	if ($str)  {
 			 	$host = split(":", $str);
 				if ($host[0]) $str = "host=".adodb_addslashes($host[0]);
-				else $str = 'host=localhost';
+				else $str = 'host='; // Modified 2004-07-23 by Kevin Jamieson (http://www.pkp.ubc.ca/)
+				// Don't specify a host if none was entered (so will connect using sockets)
 				if (isset($host[1])) $str .= " port=$host[1]";
 			}
 		   		if ($user) $str .= " user=".$user;
