--- adodb462/adodb.inc.php      Sat Apr 30 16:13:06 2005
+++ adodb/adodb.inc.php Sat Apr 30 16:11:47 2005
@@ -279,6 +279,7 @@
        var $arrayClass = 'ADORecordSet_array'; /// name of class used to generate array recordsets, which are pre-downloaded recordsets
 
        var $noNullStrings = false; /// oracle specific stuff - if true ensures that '' is converted to ' '
+       var $numQueries = 0;
        var $numCacheHits = 0; 
        var $numCacheMisses = 0;
        var $pageExecuteCountRows = true;
@@ -791,6 +792,8 @@
         */
        function &Execute($sql,$inputarr=false) 
        {
+               $this->numQueries++;
+
                if ($this->fnExecute) {
                        $fn = $this->fnExecute;
                        $ret =& $fn($this,$sql,$inputarr);
@@ -819,7 +822,7 @@
                                                else if ($typ == 'double')
                                                        $sql .= str_replace(',','.',$v); // locales fix so 1.1 does not get converted to 1,1
                                                else if ($typ == 'boolean')
-                                                       $sql = $v ? $this->true : $this->false;
+                                                       $sql .= $v ? $this->true : $this->false;
                                                else if ($v === null)
                                                        $sql .= 'NULL';
                                                else
