Index: adodb/adodb-datadict.inc.php
===================================================================
RCS file: /cvs/ojs2/lib/adodb/adodb-datadict.inc.php,v
retrieving revision 1.3
diff -u -r1.3 adodb-datadict.inc.php
--- adodb/adodb-datadict.inc.php	12 Mar 2004 23:05:13 -0000	1.3
+++ adodb/adodb-datadict.inc.php	28 Jun 2004 00:39:48 -0000
@@ -169,6 +169,7 @@
 	var $dataProvider;
 	var $blobSize = 100; 	/// any varchar/char field this size or greater is treated as a blob
 							/// in other words, we use a text area for editting.
+	var $charSet; // Added 2004-06-20 by Kevin Jamieson (http://www.pkp.ubc.ca/)
 	
 	function GetCommentSQL($table,$col)
 	{
@@ -663,5 +664,24 @@
 		
 		return $sql;
 	}
+	
+	// Functions for managing the database character encoding
+	// (for CREATE DATABASE, CREATE TABLE, etc.)
+	// Added 2004-06-20 by Kevin Jamieson (http://www.pkp.ubc.ca/)
+	function GetCharSet()
+	{
+		if (!$this->charSet) {
+			return false;
+		} else {
+			return $this->charSet;
+		}
+	}
+	
+	// SetCharSet - switch the client encoding
+	function SetCharSet($charset_name)
+	{
+		$this->charSet = $charset_name;
+	}
+	
 } // class
 ?>
\ No newline at end of file
