Index: ./datadict/datadict-postgres.inc.php
===================================================================
RCS file: /cvs/ojs2/lib/adodb/datadict/datadict-postgres.inc.php,v
retrieving revision 1.3
retrieving revision 1.5
diff -u -r1.3 -r1.5
--- ./datadict/datadict-postgres.inc.php	12 Mar 2004 23:05:13 -0000	1.3
+++ ./datadict/datadict-postgres.inc.php	24 Jul 2004 10:23:43 -0000	1.5
@@ -210,5 +210,24 @@
 		
 		return $sql;
 	}
+	
+	// Functions for managing the database character encoding
+	// (for CREATE DATABASE, CREATE TABLE, etc.)
+	// Added 2004-06-20 by Kevin Jamieson (http://www.pkp.ubc.ca/)
+	// NOTE: If a character set is specified, assumes the database server supports this
+	function CreateDatabase($dbname,$options=false)
+	{
+		$options = $this->_Options($options);
+		$sql = array();
+		
+		$s = 'CREATE DATABASE ' . $this->NameQuote($dbname);
+		if (isset($options[$this->upperName]))
+			$s .= ' '.$options[$this->upperName];
+		if ($this->charSet)
+			$s .= sprintf(' WITH ENCODING \'%s\'', $this->charSet);
+		if (7.3 <= (float) @$this->serverInfo['version'])
+			$s .= ' TEMPLATE template0'; // Deal with "template1 is being accessed by other users" errors (FIXME?)
+		$sql[] = $s;
+		return $sql;
+	}
 }
 ?>
\ No newline at end of file
